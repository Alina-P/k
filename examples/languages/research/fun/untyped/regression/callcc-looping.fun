// using callcc for looping

let goto = 0 and n = 0
in callcc (fun exit -> (
             callcc (fun k -> (& goto := k)) ;   --- "goto _" will jump here
                       if n < 100
                       then & n := n + 1
                       else exit n ;
                       goto 0
                     ) )
