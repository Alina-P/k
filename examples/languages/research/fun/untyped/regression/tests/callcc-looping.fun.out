<result>
  100
</result>
